docs/source/final_attrs.rst

```diff
@@ -0,0 +1,285 @@
+     def fun(x: Final[List[int]]) ->  None:  # Error!
+         ...
+
+* ``Final`` and ``ClassVar`` should not be used together. Mypy will infer
+  the scope of a final declaration automatically depending on whether it was
+  initialized in class body or in ``__init__``.
+
+.. note::
+   Conditional final declarations and final declarations within loops are
+   rejected.
+
+Using final attributes
+**********************
+
+As a result of a final declaration mypy strives to provide the
+two following guarantees:
+
+* A final attribute can't be re-assigned (or otherwise re-defined), both
+  internally and externally:
+
+  .. code-block:: python
+
+     # file mod.py
+     from typing_extensions import Final
+
+     ID: Final = 1
+
+     class SomeCls:
+         ID: Final = 1
+
+         def meth(self) -> None:
+             self.ID = 2  # Error: can't assign to final attribute
+
+     # file main.py
+     import mod
+     mod.ID = 2  # Error: can't assign to constant.
+
+     from mod import ID
+     ID = 2  # Also an error, see note below.
+
+     class DerivedCls(mod.SomeCls):
+         ...
+
+     DerivedCls.ID = 2  # Error!
+     obj: DerivedCls
+     obj.ID = 2  # Error!
+
+* A final attribute can't be overridden by a subclass (even with another
+  explicit final declaration). Note however, that final attributes can
+  override read-only properties. This also applies to multiple inheritance:
+
+  .. code-block:: python
+
+     class Base:
+         @property
+         def ID(self) -> int: ...
+
+     class One(Base):
+         ID: Final = 1  # OK
+
+     class Other(Base):
+         ID: Final = 2  # OK
+
+     class Combo(One, Other):  # Error: cannot override final attribute.
+         pass
+
+* Declaring a name as final only guarantees that the name wll not be re-bound
+  to other value, it doesn't make the value immutable. One can use immutable ABCs
+  and containers to prevent mutating such values:
+
+  .. code-block:: python
+
+     x: Final = ['a', 'b']
+     x.append('c')  # OK
+
+     y: Final[Sequance[str]] = ['a', 'b']
+     y.append('x')  # Error: Sequance is immutable
+     z: Final = ('a', 'b')  # Also an option
+
+.. note::
+
+   Mypy treats re-exported final names as final. In other words, once declared,
+   the final status can't be "stripped". Such behaviour is typically desired
+   for larger libraries where constants are defined in a separate module and
+   then re-exported.
+
+Final methods
+*************
+
+Like with attributes, sometimes it is useful to protect a method from
+overriding. In such situations one can use a ``typing_extensions.final``
+decorator:
+
+.. code-block:: python
+
+   from typing_extensions import final
+
+   class Base:
+       @final
+       def common_name(self) -> None:
+           ...
+
+   # 1000 lines later
+
+   class Derived(Base):
+       def common_name(self) -> None:  # Error: this overriding might break
+                                       # invariants in the base class.
```

Comment:
I think it's best to leave the explanation out, since it doesn't really add much. If the reader understands what "final" is useful for, the explanation adds nothing. If the reader doesn't understand why somebody would use `@final`, the explanation probably isn't sufficient, and may be confusing. I'd write this as `# Error: can't override a final method`.

Reply:
So `can't` or `cannot`? I assume the latter.